report_alternative_layout/models/ir_actions_report.py

```diff
@@ -19,11 +16,18 @@ class Report(models.Model):
         "Show Remit-to Bank",
         help="If selected, remit-to bank account will show in the report output.",
     )
+    show_document_number = fields.Boolean()
+    date_field_id = fields.Many2one(
+        "ir.model.fields",
+        domain="[('model','=', model), ('ttype', 'in', ('date', 'datetime'))]",
+    )
 
     def _render_qweb_pdf(self, report_ref, res_ids=None, data=None):
         report = self._get_report(report_ref)
-        if report.apply_alternative_layout:
+        if report.paperformat_id.apply_alternative_layout:
             self = self.with_context(apply_alternative_layout=True)
+        if report.paperformat_id.show_address_in_header:
+            self = self.with_context(show_address_in_header=True)
```

Comment:
Non-blocking but this should be slightly more efficient.
```suggestion
        self = self.with_context(
            apply_alternative_layout=report.paperformat_id.apply_alternative_layout,
            show_address_in_header=report.paperformat_id.show_address_in_header,
        )
```